### PR TITLE
Transformer plugin for parcel

### PR DESCRIPTION
**Pull request by SquCode**
- Created a transformer plugin for parcel: parcel-transformer-telefunc. Note: It must be published as an individual package to follow parcel plugin's naming standard
- express-and-parcel example
- export the transformer api under: ./transformer
- minor: ignore all example's package-lock files
- dev: gitignore .idea ide files

### CODE REVIEW

**Comment by brillout** [comment]
👍 Will have a look after I release vps 0.4.

Looking forward to add the Parcel transformer.

**Comment by brillout** [comment]
(I will also apply the changes to the docs we discussed then.)

**Comment by brillout** [comment]
> 👍 Will have a look after I release vps 0.4.

ETA next week.

Looking forward to review you PR :-).

**Comment by SquCode** [comment]
Ok, just take your time !

**Comment by brillout** [comment]
Alright. Let's merge this.

>  It must be published as an individual package to follow parcel plugin's naming standard

Is that strictly needed? While in the long run I'd agree that we should publish it independently, for now I'd rather include the Parcel plugin within the `telefunc` package.

File: parcel-transformer-telefunc/transformer.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const findPkg = require("find-pkg");
+
+//import {transformTelefuncFile} from "telefunc/transformer";
+const {transformTelefuncFile} = require("telefunc/transformer");
+
+export default new Transformer({
+    async transform({asset}) {
+
+        // Determine package path:
+        const packageJsonFile =await findPkg(dirname(asset.filePath));
```
Comment (brillout) [inline parcel-transformer-telefunc/transformer.ts:13]: Instead of using `findPkg()`, doesn't Parcel provide the user's app root directory?

**Comment by SquCode** [comment]
> > It must be published as an individual package to follow parcel plugin's naming standard
> 
> Is that strictly needed?

Yes, unfortunately: It's a strict requirement and also the [plugin "repository"](https://parceljs.org/plugin-browser) is just a simple pattern matching over NPM.

File: parcel-transformer-telefunc/transformer.ts (hunk 1 above)
Comment (brillout) [reply parcel-transformer-telefunc/transformer.ts:13]: https://parceljs.org/plugin-system/api/#PluginOptions => isn't `projectRoot` what we want?

**Comment by brillout** [comment]
We don't need to be included in Parcel's plugin listing.

So, what happens if we do the following?

```
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.telefunc.*": ["telefunc/parcel", "..."]
  }
}
```

**Comment by SquCode** [comment]
Hey, it's half a year ago now. Try it yourself... ;)

**Comment by brillout** [comment]
Makes sense :-).

Closing. New PR welcome.

**Pull request closed without merging**